metadata/src/main/java/org/apache/kafka/controller/PartitionChangeBuilder.java

```diff
@@ -79,29 +84,39 @@ public enum Election {
     private List<Integer> targetReplicas;
     private List<Integer> targetRemoving;
     private List<Integer> targetAdding;
+    private List<Integer> targetElr;
+    private List<Integer> targetLastKnownElr;
+    private List<Integer> uncleanShutdownReplicas;
     private Election election = Election.ONLINE;
     private LeaderRecoveryState targetLeaderRecoveryState;
     private boolean zkMigrationEnabled;
+    private boolean eligibleLeaderReplicasEnabled;
+    private int minISR;
 
 
     public PartitionChangeBuilder(
         PartitionRegistration partition,
         Uuid topicId,
         int partitionId,
         IntPredicate isAcceptableLeader,
-        MetadataVersion metadataVersion
+        MetadataVersion metadataVersion,
+        int minISR
     ) {
         this.partition = partition;
         this.topicId = topicId;
         this.partitionId = partitionId;
         this.isAcceptableLeader = isAcceptableLeader;
         this.metadataVersion = metadataVersion;
         this.zkMigrationEnabled = false;
+        this.eligibleLeaderReplicasEnabled = false;
+        this.minISR = minISR;
 
         this.targetIsr = Replicas.toList(partition.isr);
         this.targetReplicas = Replicas.toList(partition.replicas);
         this.targetRemoving = Replicas.toList(partition.removingReplicas);
         this.targetAdding = Replicas.toList(partition.addingReplicas);
+        this.targetElr = Collections.emptyList();
+        this.targetLastKnownElr = Collections.emptyList();
```

Comment:
Why not copy the arrays out of the record at this point? I think that would help us avoid the boxing. We can copy the data out from the record and just not use if if the feature is not supported or disabled.

Reply:
Sure, we can do that.